strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/providers/repository/ProxyRepositoryProvider.java

```diff
@@ -138,29 +128,38 @@ protected OutputStream getOutputStreamInternal(RepositoryPath repositoryPath)
     @Override
     public Long count(String storageId,
                       String repositoryId,
-                      Predicate predicate)
+                      RepositorySearchRequest predicate)
     {
         return hostedRepositoryProvider.count(storageId, repositoryId, predicate);
     }
 
     @Override
-    protected ArtifactEntry provideArtifactEntry(RepositoryPath repositoryPath) throws IOException
+    protected Artifact provideArtifact(RepositoryPath repositoryPath) throws IOException
     {
-        ArtifactEntry artifactEntry = super.provideArtifactEntry(repositoryPath);
-        ArtifactEntry remoteArtifactEntry = artifactEntry.getObjectId() == null ? new RemoteArtifactEntry() : (RemoteArtifactEntry) artifactEntry;
-
-        return remoteArtifactEntry;
+        Artifact artifactEntry = super.provideArtifact(repositoryPath);
+        if (artifactEntry.getNativeId() == null) {
```

Comment:
Cosmetic: Braces on new lines please. :)
```suggestion
        if (artifactEntry.getNativeId() == null)
        {
```